src/Fabulous.AST.Tests/Expressions/InterpolatedString.fs

```diff
@@ -9,91 +9,448 @@ open type Ast
+                      Expr(FillExpr(Int(12)), 1)
+                      Expr(FillExpr(Int(12)), 1) ]
+                )
 
-                //$"{12}{12}{12}"
                 InterpolatedStringExpr([ ConstantExpr("12"); ConstantExpr("12"); ConstantExpr("12") ])
                 InterpolatedStringExpr([ Int(12); Int(12); Int(12) ])
                 InterpolatedStringExpr([ "12"; "12"; "12" ])
+            }
+        }
+        |> produces
+            """
+$"{12}"
+$"{12}"
+$"{12}"
+$"{12}"
+$"{12}"
+$"{12}"
+$"{12}"
+$"{12}"
+$"{12}{12}{12}"
+$"{12}{12}{12}"
+$"{12}{12}{12}"
+$"{12}{12}{12}"
+"""
+
+    [<Fact>]
+    let ``Basic interpolation expressions with number of braces``() =
+        Oak() {
+            AnonymousModule() {
+                // $"{12}"
+                InterpolatedStringExpr([ Expr(FillExpr(Int(12)), 2) ])
+                InterpolatedStringExpr([ Expr(FillExpr(Int(12)), 3) ])
+                InterpolatedStringExpr([ Expr(FillExpr(Int(12)), 4) ])
+                InterpolatedStringExpr([ Expr(FillExpr("12"), 5) ])
+            }
+        }
+        |> produces
+            """
+$"{{12}}"
+$"{{{12}}}"
+$"{{{{12}}}}"
+$"{{{{{12}}}}}"
+"""
 
-                (*
-                //$"""{12}"""
-                InterpolatedRawStringExpr(ConstantExpr("12"))
-                InterpolatedRawStringExpr(Int(12))
-                InterpolatedRawStringExpr("12")
+    [<Fact>]
+    let ``Verbatim interpolation Aligning expressions in interpolated strings``() =
+        let source =
+            Oak() {
+                AnonymousModule() {
+                    // $"""|{"Left",-7}|{"Right",7}|"""
+                    InterpolatedStringExpr(
+                        [ Text("|")
+                          Expr(FillExpr(TupleExpr([ "Left"; "-7" ])), 1)
+                          Text("|")
+                          Expr(FillExpr(TupleExpr([ "Right"; "7" ])), 1)
+                          Text("|") ],
+                        isVerbatim = true
+                    )
+                }
+            }
 
-                // $"""{12}{12}{12}"""
-                InterpolatedRawStringExpr([ConstantExpr("12"); ConstantExpr("12"); ConstantExpr("12")])
-                InterpolatedRawStringExpr([Int(12); Int(12); Int(12)])
-                InterpolatedRawStringExpr(["12"; "12"; "12"])
+        (*
+$"""|{"Left",-7}|{"Right",7}|"""
+*)
 
-                // $$"""{12}"""
-                InterpolatedRawStringExpr("$$", ConstantExpr("12"))
-                InterpolatedRawStringExpr("$$", Int(12))
-                InterpolatedRawStringExpr("$$", "12")
+        let res = Gen.mkOak source |> Gen.run
 
-                // $$"""{12}{12}{12}"""
-                InterpolatedRawStringExpr("$$", [ConstantExpr("12"); ConstantExpr("12"); ConstantExpr("12")])
-                InterpolatedRawStringExpr("$$", [Int(12); Int(12); Int(12)])
-                InterpolatedRawStringExpr("$$", ["12"; "12"; "12"])
-            *)
+        Assert.NotNull(res)
```

Comment:
You could have a file that has the `$"""|{"Left",-7}|{"Right",7}|"""` and write out the actual output to a file. Then compare both files instead of having try to trick the parser into what you want.

Reply:
This might too heavy at the moment but https://github.com/VerifyTests/Verify is the same concept.